kiwi/xml_description.py

```diff
@@ -52,9 +54,9 @@ class XMLDescription(object):
         path to base XML description file
 
     """
-    def __init__(self, description, derived_from=None):
+    def __init__(self, description=None, derived_from=None, content=None):
         self.description_xslt_processed = NamedTemporaryFile()
-        self.description = description
+        self.description = content and BytesIO(bytes(content)) or description
```

Comment:
@schaefi Aaah more clear IMHO. There are some bits that can help too:

* Raise an exception when both description and content are provided. The prevalence of one over another at least needs to be documented in the header (also missing)
* I am a bit worried with the type mangling. `content` now expect bytes and can misbehave in unicode. The use of `six` indicates me that is expected to work on Py2 and Py3 so this can be a problem. Can be helpful to add test cases for unicode and str (py2) and str (py3)

I'm sure that @isbm had good reasons to use bytes (I asked to understand them), but I am curious why not use `ET.fromstring` or `ET.parse` directly to manage file and string cases.

One note also, `from builtins import bytes` is not going to work in py2, but maybe this test is only executed in Py3